torch/distributed/_sharded_tensor/api.py

```diff
@@ -751,3 +928,122 @@ def _create_tensor_from_params(*size, local_device, tensor_init_params: TensorIn
+        >>> tensor = torch.stack([tensor, tensor])
+        >>> tensor
+        tensor([[1, 2], [1, 2]]) # Rank 0
+        tensor([[3, 4], [3, 4]]) # Rank 1
+        >>> partial_tensor = _PartialTensor(tensor, distributed_c10d.ReduceOp.MAX)
+        >>> sharding_dim = 0
+        >>> collect_spec = ChunkShardingSpec(
+                dim=sharding_dim,
+                placements=[
+                    "rank:0/cuda:0",
+                    "rank:1/cuda:1",
+                ],
+            )
+        >>> complete_tensor = partial_tensor.reshard(collect_spec)
+        >>> complete_tensor
+        tensor([3, 4]) # Rank 0
+        tensor([3, 4]) # Rank 1
+
+        >>> # All tensors below are of torch.cfloat type.
+        >>> # We have 2 process groups, 2 ranks.
+        >>> tensor = torch.tensor([1+1j, 2+2j], dtype=torch.cfloat) + 2 * rank * (1+1j)
+        >>> tensor = torch.stack([tensor, tensor])
+        >>> tensor
+        tensor([[1.+1.j, 2.+2.j], [1.+1.j, 2.+2.j]]) # Rank 0
+        tensor([[3.+3.j, 4.+4.j], [3.+3.j, 4.+4.j]]) # Rank 1
+        >>> partial_tensor = _PartialTensor(tensor)
+        >>> complete_tensor = partial_tensor.reshard(collect_spec)
+        >>> complete_tensor
+        tensor([4.+4.j, 6.+6.j]) # Rank 0
+        tensor([4.+4.j, 6.+6.j]) # Rank 1
+    """
+
+    def __init__(self, local_partial_shard, pg, aggr_op=distributed_c10d.ReduceOp.SUM):
+        self.local_partial_shard = local_partial_shard
+        self.pg = pg
+        self.aggr_op = aggr_op
+
+    def __post_init__(self):
+        if not isinstance(self.local_partial_shard, torch.Tensor):
+            raise ValueError("local_partial_shard needs to be a Tensor.")
+        if not isinstance(self.aggr_op, distributed_c10d.ReduceOp):
+            raise ValueError(
+                "aggr_op needs to be a member of distributed_c10d.ReduceOp."
+            )
+
+    def reshard(self, resharding_spec: ShardingSpec) -> ShardedTensor:
+        """
+        Consolidate all partial tensors across multiple ranks and covert to a sharded
+        tensor based on the resharding spec.
+
+        Args:
+            resharding_spec (:class:`torch.distributed._sharding_spec.ShardingSpec`):
+                The specification describing how we reshard the aggregated local result.
+
+        Returns:
+            A :class:`ShardedTensor` filled with local aggregated result.
+        """
+        if not isinstance(resharding_spec, ChunkShardingSpec):
+            raise NotImplementedError("Only ChunkShardingSpec supported for reshard.")
+        sharding_dim = int(resharding_spec.dim)  # type: ignore[attr-defined]
+        world_size = dist.get_world_size(self.pg)
+        local_shards = torch.tensor_split(self.local_partial_shard, world_size)
+        local_result = reduce_scatter(
+            torch.empty_like(local_shards[0]), list(local_shards), op=self.aggr_op
+        )
+
+        rank = dist.get_rank(self.pg)
```

Comment:
`self.pg.rank` should also work:

https://github.com/pytorch/pytorch/blob/b36b11cbc1f11fa805894927ca667443f3618a34/torch/csrc/distributed/c10d/init.cpp#L964-L965

Reply:
Got it. Thanks!!